File: sentimental-readability/readability.py
from cs50 import get_string
# TODO

# for user input
user_input = get_string("Text: ")

l = 0
w = 1
s = 0
# Loop over user_input and get the number of l, w and s in the user_input
for character in user_input:
    if character.islower() or character.isupper():
        l += 1
    if character == ' ':
        w += 1
    if character == '?' or character == '!' or character == '.':
        s += 1

# Letters and Sentences per 100 w
L = l * 100 / w
S = s * 100 / w

# Coleman-Liau formula
grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")

File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    // Get user input
    string u_input = get_string("Text: ");

    float sentences = 0;
    float words = 1; // Initialized to 1 otherwise it will not count the 1st word
    float letters = 0;

    // Iterate over the user input
    for (int i = 0; i < strlen(u_input); i++)
    {
        
        if (u_input[i] == '!' || u_input[i] == '.' || u_input[i] == '?')
        {
            sentences++;
        }

        if (u_input[i] == ' ')
        {
            words++;
        }

        if (islower(u_input[i]) || isupper(u_input[i]))
        {
            letters++;
        }
    }

    float L = 100 * letters / words;   // L = Number of letters per 100 words
    float S = 100 * sentences / words; // S = Number of sentences per 100 words

    // Formula for evaluating grade
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }

    return 0;
}